OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -534,5 +534,16 @@ public IEnumerable<Actor> ActorsInWorld()
 		{
 			return bins.SelectMany(bin => bin.Actors.Where(actor => actor.IsInWorld));
 		}
+
+		public IEnumerable<Actor> GetActorsOnRay(CPos start, CPos end)
```

Comment:
You should be able to simplify this down to something like `return Raycaster.Raycast(start, end).SelectMany(c => GetUnitsAt(c)).Distinct();` (note: the distinct is important to stop you from picking up buildings multiple times).